src/main/java/net/sf/jabref/logic/importer/fetcher/MedlineFetcher.java

```diff
@@ -0,0 +1,199 @@
+package net.sf.jabref.logic.importer.fetcher;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.MalformedURLException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.net.URLConnection;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import net.sf.jabref.logic.help.HelpFile;
+import net.sf.jabref.logic.importer.FetcherException;
+import net.sf.jabref.logic.importer.IdBasedFetcher;
+import net.sf.jabref.logic.importer.ParserResult;
+import net.sf.jabref.logic.importer.SearchBasedFetcher;
+import net.sf.jabref.logic.importer.fileformat.MedlineImporter;
+import net.sf.jabref.logic.l10n.Localization;
+import net.sf.jabref.model.entry.BibEntry;
+import net.sf.jabref.model.strings.StringUtil;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.http.client.utils.URIBuilder;
+
+/**
+ * Fetch or search from Pubmed http://www.ncbi.nlm.nih.gov/sites/entrez/
+ */
+public class MedlineFetcher implements IdBasedFetcher, SearchBasedFetcher {
+
+    private static final Log LOGGER = LogFactory.getLog(MedlineFetcher.class);
+
+    private static final String API_MEDLINE_FETCH = "http://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi";
+    private static final String API_MEDLINE_SEARCH = "http://eutils.ncbi.nlm.nih.gov/entrez/eutils/esearch.fcgi";
+
+    private static final Pattern ID_PATTERN = Pattern.compile("<Id>(\\d+)</Id>");
+    private static final Pattern COUNT_PATTERN = Pattern.compile("<Count>(\\d+)<\\/Count>");
+    private static final Pattern RET_MAX_PATTERN = Pattern.compile("<RetMax>(\\d+)<\\/RetMax>");
+    private static final Pattern RET_START_PATTERN = Pattern.compile("<RetStart>(\\d+)<\\/RetStart>");
+
+    /**
+     * Gets the initial list of ids
+     */
+    private SearchResult getIds(String term, int start, int pacing) {
```

Comment:
As far as I can see, only the returned IDs are used and not the additional data in SearchResult. Could you change the result to List<Integer>. Please also rename term -> query